flink-runtime/src/main/java/org/apache/flink/runtime/state/AbstractKeyedStateBackend.java

```diff
@@ -250,6 +250,13 @@ public void setCurrentKey(K newKey) {
                 KeyGroupRangeAssignment.assignToKeyGroup(newKey, numberOfKeyGroups));
     }
 
+    /** Only used in {@code AsyncKeyedStateBackendAdaptor}. */
+    public void setCurrentKeyAndKeyGroupIndex(K newKey, int newKeyGroupIndex) {
```

Comment:
Can we add `setCurrentKeyAndKeyGroupIndex`  to `KeyedStateBackend` interface?

nit: let `setCurrentKey` use the common part of `setCurrentKeyAndKeyGroupIndex` ?

Reply:
good point, will add soon